### DISCUSSION TITLE
Is it possible to make a field of type JSON

### DISCUSSION THREAD

**Discussion by cyr-ius** [fastapi/sqlmodel#1105] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlalchemy.dialects.sqlite import JSON
from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column

class Base(DeclarativeBase):
    pass

class Settings(Base):
    __tablename__ = "settings"
    id: Mapped[int] = mapped_column(primary_key=True)
    data: Mapped[JSON] = mapped_column(type_=JSON)
```

### Description

I would like to know how it is possible to convert a JSON type field with SQLModel

### Operating System

Linux

### SQLModel Version

0.0.22

### Python Version

3.12

**Comment by Meepoljdx** [comment, marked as answer]
```python
from sqlmodel import Field, Column, SQLModel
from sqlalchemy.dialects.postgresql import JSONB

class AlertBase(SQLModel):
    name: str = Field(nullable=False, description="The name of alert", max_length=56)
    title: str = Field(nullable=True, description="The title of alert", max_length=100)
    labels: dict = Field(sa_column=Column(JSONB))
```

**Comment by monchin** [comment]
`SQLModel` class is a derived class of pydantic `BaseModel` class, so you must let the types of the attributes in your model  be able to convert to a json, which means the types (left of the equals) could only be str, int, float, dict, list, None... , and `Map` is not allowed. But you can set sqlalchemy types (right of the equals) in the `Field` by `sa_column`/`sa_type`...
